drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -9,21 +9,32 @@ class SubmitForm < BaseMutation
     description 'Submit a form to create/update HUD record(s)'
 
     argument :input, Types::HmisSchema::FormInput, required: true
+    argument :record_lock_version, Integer, required: false
 
     field :record, Types::HmisSchema::SubmitFormResult, null: true
 
-    def resolve(input:)
+    def resolve(input:, record_lock_version: nil)
       # Look up form definition
       definition = Hmis::Form::Definition.find_by(id: input.form_definition_id)
       raise HmisErrors::ApiError, 'Form Definition not found' unless definition.present?
 
+      # FIXME: locking around the form definition isn't right. Probably want to lock on record or record.owner
```

Comment:
Agreed. I don't think there is any need to lock the definition, we're not updating it at all.

Reply:
The lock here is intended to prevent submissions for the same definition from running in parallel. The definition is convenient for this since it's guaranteed to exist and will be the same across mutations; it's essentially functioning as an advisory lock here so I don't think it matters that the definition isn't updated. But yes, it is a bit of a hack.

Perhaps I should build something that infers the correct record to lock on from the input or we could use an actual advisory lock.

Reply:
> The lock here is intended to prevent submissions for the same definition from running in parallel.

We definitely don't want that. That would mean that no two intake assessments cross the entire system could be submitted at the same time. Or, in the case for SubmitForm, it would prevent 2 services from being recorded at the same time (for example).

Reply:
Per discussion on slack - the definition lock shouldn't have much practical impact at our current small scale. However it feels like a hack so I'll replace with simple tx for now; we can rely on optimistic locking for the record types that support it. If integrity issues surface we can re-examine this